components/brave_search/browser/brave_search_default_host.cc

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "brave/components/brave_search/browser/brave_search_default_host.h"
+
+#include <string>
+#include <utility>
+
+#include "brave/components/brave_search/browser/prefs.h"
+#include "brave/components/weekly_storage/daily_storage.h"
+#include "components/prefs/pref_service.h"
+#include "components/search_engines/template_url.h"
+#include "components/search_engines/template_url_service.h"
+
+namespace {
+
+TemplateURL* GetSearchTemplateForSite(TemplateURLService* service,
+                                      const std::string& host) {
+  // Prefer built-in entries, then offer site-provided entries,
+  // but ignore extension-provided entries.
+  TemplateURL::TemplateURLVector urls = service->GetTemplateURLs();
```

Comment:
@simonhong do we need any null checks here?

Reply:
|urls| only includes valid providers.
Upstream also iterates urls like below from many places.
```
for (const auto& turl : template_urls_) {
    if (turl->xxx())
```
```